File: ts/1024/977.ts
function sortedSquares(nums: number[]): number[] {
  const n = nums.length, ans: number[] = new Array(n)
  let i = 0, j = n - 1, l = j
  while (i <= j) {
    let x = nums[i] * nums[i], y = nums[j] * nums[j]
    if (x >= y) {
      ans[l] = x
      i += 1
    } else {
      ans[l] = y
      j -= 1
    }
    l -= 1
  }
  return ans
};

File: c/1024/977.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortedSquares(int* nums, int numsSize, int* returnSize){
  int l = 0, r = numsSize - 1, idx = r;
  int* ans = malloc(sizeof(int) * numsSize);
  *returnSize = numsSize;

  while (l <= r) {
    int x = nums[l] * nums[l], y = nums[r] * nums[r];
    if (x >= y) {
      ans[idx--] = x;
      l += 1;
    } else {
      ans[idx--] = y;
      r -= 1;
    }
  }

  return ans;
}